migrations/20230116124310_alter_phone_type.up.sql

```diff
@@ -1,5 +1,14 @@
 -- alter phone field column type to accomodate for soft deletion 
 
-alter table auth.users
-alter column phone type text,
-alter column phone_change type text;
+do $$
+begin
+  alter table {{ index .Options "Namespace" }}.users
+    alter column phone type text,
+    alter column phone_change type text;
+exception
+  -- SQLSTATE errcodes https://www.postgresql.org/docs/current/errcodes-appendix.html
+  when SQLSTATE '0A000' then
+    raise notice 'Unable to change data type of phone, phone_change columns due to use by a view or rule';
+  when SQLSTATE '2BP01' then
+    raise notice 'Unable to change data type of phone, phone_change columns due to dependent objects';
```

Comment:
if it raises a different error code, what would happen? also, just to confirm, if there is an exception that and it falls into `raise notice`, would the migration still "succeed" and get inserted into the `auth.schema_migrations` table? if it's inserted into the `auth.schema_migrations` table, is there a way to prevent that from happening so we always attempt to run this migration on deploy / restart.

Reply:
There's no way to retry it each time. Maybe in the discussion later we can talk about separating migrations into stages. I'd rather not busy the `User` model with the obfuscation stuff, it's only used here.

Reply:
The notice will just emit a message 

> The level option specifies the error severity. Allowed levels are DEBUG, LOG, INFO, NOTICE, WARNING, and EXCEPTION, with EXCEPTION being the default. EXCEPTION raises an error (which normally aborts the current transaction); the other levels only generate messages of different priority levels. Whether messages of a particular priority are reported to the client, written to the server log, or both is controlled by the [log_min_messages](https://www.postgresql.org/docs/current/runtime-config-logging.html#GUC-LOG-MIN-MESSAGES) and [client_min_messages](https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-CLIENT-MIN-MESSAGES) configuration variables. See [Chapter 20](https://www.postgresql.org/docs/current/runtime-config.html) for more information.

Reply:
hmm ok so we will also need some way to keep track of these projects that have emitted this notice so that we can rerun the migration for them eventually